scripts/mk_configuration.py

```diff
@@ -154,11 +152,15 @@ def config_for_board(top, flows_list, build_dir):
 # define_name: name of the macro in confs (its called define because it is unvalued, it is either set or unset)
 # should_set: Select if define should be set or not
 def update_define(confs, define_name, should_set):
-    for idx, macro in enumerate(confs):
+    for macro in confs:
         if macro['name']==define_name:
             # Found macro. Unset it if not 'should_set'
-            if not should_set: del confs[idx]
+            if should_set: 
+                macro['val'] = '1'
+            else:
+                confs.remove(macro)
```

Comment:
Maybe it should be set o NA instead of removing it, to be consistent with `if macro['val'] != 'NA':` of mk_configuration.py
```suggestion
                macro['val'] = 'NA'
```